### DISCUSSION TITLE
Constraint at Final Time Step in Inverse Problem

### DISCUSSION THREAD

**Discussion by ppandit95** [lululxvi/deepxde#1475] [Q&A]
Dear Dr. @lululxvi 

I'm solving an inverse problem to learn coefficients in diffusion equation with (x,t) as input and c as output, but I wish to implement a constraint involving coefficients of PDE only at the final time step and for all positions. So any headers in this regard will prove beneficial .

Thanks in Advance
Pushkar

**Comment by lululxvi** [comment]
Exactly the same approach as the examples.

**Reply by ppandit95** [reply]
Firstly, Thanks, Dr. @lululxvi for the resolution of the above query, but can you refer to a specific example wherein constraint is incorporated at a particular time . It will immensely help to develop the model.

Many thanks for considering my request.
Pushkar

**Reply by ppandit95** [reply]
Thanks Alot  Dr. @lululxvi for helping me out there. As you suggested I have implemented the constraint at final time using the statement as - 
`constraint2 = dde.OperatorBC(geomtime,diffusivity_constraint,lambda x,_: dde.utils.isclose(x[0],0.0) and dde.utils.isclose(x[1],1.0))`

But in here, although I can incorporate constraint at any value of x[0] within the domain but x[1] values only work for the end points in the time domain. Any suggestions on this because otherwise if i try to implement constraint at x[1]=0.75 , I m getting an error as - 
`
Traceback (most recent call last):
  File "/home/pushkar/DeepXDE/WorkSpace/Binary_Diffusion/ManufacturedSolution/NormalizedManufacturedSolution/LateStartSimulation/manufactured_sol_normalized.py", line 116, in <module>
    dde.saveplot(losshistory, train_state, issave=True, isplot=True)
  File "/home/pushkar/DeepXDE/lib/python3.10/site-packages/deepxde/utils/external.py", line 175, in saveplot
    save_best_state(train_state, train_fname, test_fname)
  File "/home/pushkar/DeepXDE/lib/python3.10/site-packages/deepxde/utils/external.py", line 344, in save_best_state
    test = np.hstack((train_state.X_test, best_y))
  File "<__array_function__ internals>", line 200, in hstack
  File "/home/pushkar/DeepXDE/lib/python3.10/site-packages/numpy/core/shape_base.py", line 370, in hstack
    return _nx.concatenate(arrs, 1, dtype=dtype, casting=casting)
  File "<__array_function__ internals>", line 200, in concatenate
ValueError: all the input arrays must have same number of dimensions, but the array at index 0 has 2 dimension(s) and the array at index 1 has 1 dimension(s)
`
I appreciate any help you can provide.
 
Pushkar

**Reply by ppandit95** [reply]
Basically, what I want is to implement a constraint at a specific position and time within the scope of simulation . Any help in this regard will prove beneficial.

Thanks in Advance
Pushkar

**Comment by DK-fish** [comment]
收到，感谢你（们）的祝福！

**Comment by ppandit95** [comment]
Dear Dr. @lululxvi

I am still facing the issue of implementing constraint at a specific position and specific time while solving the time-dependent pde inverse problem . Any help in this regard will prove beneficial.

Thanks in Advance
Pushkar
